spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/tracing/zipkin/ZipkinConfigurationsBraveConfigurationTests.java

```diff
@@ -40,61 +46,126 @@
 class ZipkinConfigurationsBraveConfigurationTests {
 
 	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
-		.withConfiguration(AutoConfigurations.of(BraveConfiguration.class));
+		.withConfiguration(AutoConfigurations.of(DefaultEncodingConfiguration.class, BraveConfiguration.class));
 
 	private final ApplicationContextRunner tracingDisabledContextRunner = this.contextRunner
 		.withPropertyValues("management.tracing.enabled=false");
 
 	@Test
 	void shouldSupplyBeans() {
-		this.contextRunner.withUserConfiguration(ReporterConfiguration.class)
-			.run((context) -> assertThat(context).hasSingleBean(ZipkinSpanHandler.class));
+		this.contextRunner.withUserConfiguration(SenderConfiguration.class)
+			.run((context) -> assertThat(context).hasSingleBean(AsyncZipkinSpanHandler.class));
 	}
 
 	@Test
 	void shouldNotSupplySpanHandlerIfReporterIsMissing() {
-		this.contextRunner.run((context) -> assertThat(context).doesNotHaveBean(ZipkinSpanHandler.class));
+		this.contextRunner.run((context) -> assertThat(context).doesNotHaveBean(AsyncZipkinSpanHandler.class));
 	}
 
 	@Test
 	void shouldNotSupplyIfZipkinReporterBraveIsNotOnClasspath() {
+		// Note: Technically, Brave can work without zipkin-reporter. For example,
```

Comment:
Looking at `org.springframework.boot.actuate.autoconfigure.tracing.zipkin.ZipkinConfigurations.BraveConfiguration`, it provides `BytesEncoder<MutableSpan>` and `AsyncZipkinSpanHandler`, both are in the `zipkin-reporter` dependency. 

You mean removing the zipkin dependency from Brave could resolve https://github.com/micrometer-metrics/tracing/issues/509?

Reply:
So, if things are same as when I wrote it, wavefront doesn't use any of the zipkin-reporter types, just brave's `brave.handler.SpanHandler`. Means its only dependency is brave's core jar.

Practically speaking, all others will need zipkin-reporter-brave, which depends on zipkin-reporter, but not zipkin's core jar (anymore) as BytesEncoder<MutableSpan> and AsyncZipkinSpanHandler do not need those types.

```xml
    <dependency>
      <groupId>${project.groupId}</groupId>
      <artifactId>zipkin-reporter</artifactId>
      <version>${project.version}</version>
      <!-- Senders don't use zipkin types. Excluding allows brave users to
           avoid them by default. -->
      <exclusions>
        <exclusion>
          <groupId>io.zipkin.zipkin2</groupId>
          <artifactId>zipkin</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

That said, the otel exporter *does* use io.zipkin.zipkin2:zipkin, for AsyncReporter<zipkin2.Span> and correspondingly `BytesEncoder<MutableSpan>`

hope these notes help. I'm knackered from several very long days so will get back to more feedback tomorrow asia time!

Reply:
this discussion is something that can be done in another PR. I'm glad the comment drew attention to it, but it isn't super critical as zipkin-reporter-brave jar is only a few 10kb